Existing `flake8` errors in source and tests

There are existing flake8 lint warnings in the source and tests
```
(.venv) > flake8 {pandas_vet,tests}
pandas_vet/__init__.py:116:80: E501 line too long (83 > 79 characters)
pandas_vet/__init__.py:134:80: E501 line too long (94 > 79 characters)
pandas_vet/__init__.py:137:80: E501 line too long (83 > 79 characters)
pandas_vet/__init__.py:155:80: E501 line too long (105 > 79 characters)
pandas_vet/__init__.py:158:80: E501 line too long (83 > 79 characters)
pandas_vet/__init__.py:174:80: E501 line too long (107 > 79 characters)
pandas_vet/__init__.py:177:80: E501 line too long (83 > 79 characters)
pandas_vet/__init__.py:232:80: E501 line too long (85 > 79 characters)
pandas_vet/__init__.py:365:80: E501 line too long (84 > 79 characters)
pandas_vet/__init__.py:369:80: E501 line too long (82 > 79 characters)
pandas_vet/__init__.py:373:80: E501 line too long (84 > 79 characters)
pandas_vet/__init__.py:383:80: E501 line too long (83 > 79 characters)
pandas_vet/__init__.py:386:80: E501 line too long (85 > 79 characters)
pandas_vet/__init__.py:389:80: E501 line too long (102 > 79 characters)
pandas_vet/__init__.py:392:80: E501 line too long (93 > 79 characters)
pandas_vet/__init__.py:395:80: E501 line too long (90 > 79 characters)
pandas_vet/__init__.py:398:80: E501 line too long (81 > 79 characters)
pandas_vet/__init__.py:401:80: E501 line too long (107 > 79 characters)
tests/test_PD015.py:56:80: E501 line too long (86 > 79 characters)
```
The simple fix here is to just update this code.

But it may also be a good idea to add `flake8` lint check it CI, and to consider the library's interest in adopting black auto-formatting and how that might influence a `flake8` config file.

--- a/.flake8
+++ b/.flake8
@@ -1,0 +1,2 @@
+[flake8]
+max-line-length = 100

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,14 +6,28 @@ sudo: false
 # python settings
 language: python
 python:
-  - "3.5"
-  - "3.6"
-  - "3.7"
   - "3.8"
+  - "3.7"
+  - "3.6"
+  - "3.5"
+
+# for use in linting and static analysis
+env:
+  - SOURCES="pandas_vet setup.py tests --exclude tests/data"
+
+# include allows has the following jobs run only against the first
+# entry in the python array. We use this because we both black and
+# flake8 only need to be run across the sources once.
+jobs:
+  include:
+    - install: pip install -r reqs/lint.txt
+      script: flake8 $SOURCES
+    - install: pip install -r reqs/lint.txt
+      script: black --check $SOURCES
 
 # install packages
 install:
-  - pip install -r requirements_test.txt
+  - pip install -r reqs/test.txt
   - pip install -e .
 
 # run test

--- a/README.md
+++ b/README.md
@@ -68,22 +68,30 @@ Because this project started during the PyCascades 2019 sprints, we adopt the Py
 
 3. Get a development environment set up with your favorite environment manager (`conda`, `virtualenv`, etc.). 
 
-    1. You can create one from `pip install -r requirements_test.txt` or, if you use Docker, you can build an image from the Dockerfile included in this repo.
+    0. You must use at least python 3.6 to develop, for [black](https://github.com/psf/black) support.
+
+    1. You can create one from `pip install -r requirements_dev.txt` or, if you use Docker, you can build an image from the Dockerfile included in this repo.
 
     2. Once your enviroment is set up you will need to install pandas-vet in development mode. Use `pip install -e .` (use this if you are alreay in your virtual enviroment) or `pip install -e <path>` (use this one if not in the virtual enviroment and prefer to state explicitly where it is going).
 
 
 4. Write code, docs, etc.
 
-5. We use `pytest` and `flake8` to validate our codebase. The TravisCI integration will complain on pull requests if there are any failing tests or lint violations. To check these locally, run the following commands:
+5. We use `pytest`, `flake8`, and `black` to validate our codebase. TravisCI integration will complain on pull requests if there are any failing tests or lint violations. To check these locally, run the following commands:
 
 ```bash
-pytest tests
+pytest --cov=pandas_vet
 ```
 
 ```bash
 flake8 pandas_vet setup.py tests --exclude tests/data
 ```
+
+```bash
+black --check pandas_vet setup.py tests --exclude tests/data
+```
+
+
 
 6. Push to your forked repo.
 

--- a/pandas_vet/__init__.py
+++ b/pandas_vet/__init__.py
@@ -201,10 +201,7 @@ def check_for_arithmetic_methods(node: ast.Call) -> List:
         "mod",
     ]
 
-    if (
-        isinstance(node.func, ast.Attribute)
-        and node.func.attr in arithmetic_methods
-    ):
+    if isinstance(node.func, ast.Attribute) and node.func.attr in arithmetic_methods:
         return [PD005(node.lineno, node.col_offset)]
     return []
 
@@ -217,10 +214,7 @@ def check_for_comparison_methods(node: ast.Call) -> List:
     """
     comparison_methods = ["gt", "lt", "ge", "le", "eq", "ne"]
 
-    if (
-        isinstance(node.func, ast.Attribute)
-        and node.func.attr in comparison_methods
-    ):
+    if isinstance(node.func, ast.Attribute) and node.func.attr in comparison_methods:
         return [PD006(node.lineno, node.col_offset)]
     return []
 

--- a/pandas_vet/__init__.py
+++ b/pandas_vet/__init__.py
@@ -152,7 +152,8 @@ def check_for_isnull(node: ast.Call) -> List:
     Approved:
         df.isna()
 
-    Error/warning message to recommend usage of .isna() instead of .isnull(). Functionality is equivalent
+    Error/warning message to recommend usage of .isna() instead of .isnull().
+    Functionality is equivalent.
 
     :param node: an AST node of type Call
     :return errors: list of errors of type PD003 with line number and column offset
@@ -171,7 +172,8 @@ def check_for_notnull(node: ast.Call) -> List:
     Approved:
         df.notna()
 
-    Error/warning message to recommend usage of .notna() instead of .notnull(). Functionality is equivalent
+    Error/warning message to recommend usage of .notna() instead of .notnull().
+    Functionality is equivalent.
 
     :param node: an AST node of type Call
     :return errors: list of errors of type PD004 with line number and column offset
@@ -380,7 +382,8 @@ PD009 = VetError(
     message="PD009 Use '.iloc' instead of '.iat'.  If speed is important, use numpy."
 )
 PD010 = VetError(
-    message="PD010 '.pivot_table' is preferred to '.pivot' or '.unstack'; provides same functionality"
+    message="PD010 '.pivot_table' is preferred to '.pivot' or '.unstack'; "
+    "provides same functionality"
 )
 PD011 = VetError(
     message="PD011 Use '.array' or '.to_array()' instead of '.values'; 'values' is ambiguous"
@@ -392,7 +395,8 @@ PD013 = VetError(
     message="PD013 '.melt' is preferred to '.stack'; provides same functionality"
 )
 PD015 = VetError(
-    message="PD015 Use '.merge' method instead of 'pd.merge' function. They have equivalent functionality."
+    message="PD015 Use '.merge' method instead of 'pd.merge' function. "
+    "They have equivalent functionality."
 )
 
 PD901 = VetError(

--- a/reqs/README.md
+++ b/reqs/README.md
@@ -1,0 +1,7 @@
+# `pandas-vet` Requirements
+
+This directory splits out requirements for different build, lint, & test related activities.
+
+This is mostly to speed up and dedupe travis jobs, which individually may only run part of the build.
+
+If you're developing `pandas-vet` follow the instructions in the master readme.

--- a/reqs/README.md
+++ b/reqs/README.md
@@ -2,6 +2,6 @@
 
 This directory splits out requirements for different build, lint, & test related activities.
 
-This is mostly to speed up and dedupe travis jobs, which individually may only run part of the build.
+This is mostly to speed up and dedupe TravisCI jobs, which individually may only run part of the build.
 
 If you're developing `pandas-vet` follow the instructions in the master README.

--- a/reqs/README.md
+++ b/reqs/README.md
@@ -4,4 +4,4 @@ This directory splits out requirements for different build, lint, & test related
 
 This is mostly to speed up and dedupe travis jobs, which individually may only run part of the build.
 
-If you're developing `pandas-vet` follow the instructions in the master readme.
+If you're developing `pandas-vet` follow the instructions in the master README.

--- a/reqs/lint.txt
+++ b/reqs/lint.txt
@@ -1,0 +1,20 @@
+# flake8, top level
+flake8==3.7.9
+
+# flake8 toolchain, transitive dependencies
+entrypoints==0.3
+mccabe==0.6.1
+pluggy==0.13.1
+pycodestyle==2.5.0
+pyflakes==2.1.1
+
+# black, top level
+black==19.10b0
+
+# black toolchain, transitive dependencies
+appdirs==1.4.3
+pathspec==0.7.0
+Click==7.0
+regex==2020.1.8
+typed-ast==1.4.1
+toml==0.10.0

--- a/reqs/test.txt
+++ b/reqs/test.txt
@@ -20,23 +20,3 @@ six==1.14.0
 wcwidth==0.1.8
 zipp==2.2.0
 
-# lint toolchain, top level
-flake8==3.7.9
-
-# lint toolchain, transitive dependencies
-entrypoints==0.3
-mccabe==0.6.1
-pluggy==0.13.1
-pycodestyle==2.5.0
-pyflakes==2.1.1
-
-# black toolchain
-black==19.10b0
-
-# black toolchain, transitive dependencies
-appdirs==1.4.3
-pathspec==0.7.0
-Click==7.0
-regex==2020.1.8
-typed-ast==1.4.1
-toml==0.10.0

--- a/reqs/test.txt
+++ b/reqs/test.txt
@@ -18,5 +18,6 @@ py==1.8.1
 pyparsing==2.4.6
 six==1.14.0
 wcwidth==0.1.8
-zipp==2.2.0
+# zipp > 1.1.0 doesn't support python 3.5, this can be dumped once 3.5 support is dropped
+zipp==1.1.0
 

--- a/requirements_dev.txt
+++ b/requirements_dev.txt
@@ -1,0 +1,2 @@
+-r reqs/test.txt
+-r reqs/lint.txt

--- a/requirements_test.txt
+++ b/requirements_test.txt
@@ -1,4 +1,21 @@
--r requirements.txt
-pytest>4.0.0
-pytest-cov>=2.6.1
+# build toolchain, from virtualenv/python
+pip==20.0.2
+setuptools==45.2.0
+wheel==0.34.2
 
+# test toolchain, top level
+pytest==5.3.5
+pytest-cov==2.8.1
+
+# test toolchain, transitive dependencies
+attrs==19.3.0
+coverage==5.0.3
+importlib-metadata==1.5.0
+more-itertools==8.2.0
+packaging==20.1
+pluggy==0.13.1
+py==1.8.1
+pyparsing==2.4.6
+six==1.14.0
+wcwidth==0.1.8
+zipp==2.2.0

--- a/requirements_test.txt
+++ b/requirements_test.txt
@@ -29,3 +29,14 @@ mccabe==0.6.1
 pluggy==0.13.1
 pycodestyle==2.5.0
 pyflakes==2.1.1
+
+# black toolchain
+black==19.10b0
+
+# black toolchain, transitive dependencies
+appdirs==1.4.3
+pathspec==0.7.0
+Click==7.0
+regex==2020.1.8
+typed-ast==1.4.1
+toml==0.10.0

--- a/requirements_test.txt
+++ b/requirements_test.txt
@@ -19,3 +19,13 @@ pyparsing==2.4.6
 six==1.14.0
 wcwidth==0.1.8
 zipp==2.2.0
+
+# lint toolchain, top level
+flake8==3.7.9
+
+# lint toolchain, transitive dependencies
+entrypoints==0.3
+mccabe==0.6.1
+pluggy==0.13.1
+pycodestyle==2.5.0
+pyflakes==2.1.1

